mmocr/datasets/preparers/config_generator.py

```diff
@@ -0,0 +1,308 @@
+                        'Please use different dataset_postfix to avoid '
+                        'conflict.')
+                self.anns[key] = ann_dict
+
+    def __call__(self) -> None:
+        """Generates the base dataset config."""
+
+        dataset_config = self._gen_dataset_config()
+
+        cfg_path = osp.join(self.config_path, self.task, '_base_', 'datasets',
+                            f'{self.dataset_name}.py')
+        if osp.exists(cfg_path):
+            while True:
+                c = input(f'{cfg_path} already exists, overwrite? (Y/n) ') \
+                    or 'Y'
+                if c.lower() == 'y':
+                    break
+                if c.lower() == 'n':
+                    return
+        mkdir_or_exist(osp.dirname(cfg_path))
+        with open(cfg_path, 'w') as f:
+            f.write(
+                f'{self.dataset_name}_{self.task}_data_root = \'{self.data_root}\'\n'  # noqa: E501
+            )
+            f.write(dataset_config)
+
+    @abstractmethod
+    def _gen_dataset_config(self) -> str:
+        """Generate a full dataset config based on the annotation file
+        dictionary.
+
+        Returns:
+            str: The generated dataset config.
+        """
+
+
+@CFG_GENERATORS.register_module()
+class TextDetConfigGenerator(BaseConfigGenerator):
+    """Text detection config generator.
+
+    Args:
+        data_root (str): The root path of the dataset.
+        dataset_name (str): The name of the dataset.
+        train_anns (List[Dict], optional): A list of train annotation files
+            to appear in the base configs. Defaults to
+            ``[dict(file='textdet_train.json')]``.
+            Each element is typically a dict with the following fields:
+            - ann_file (str): The path to the annotation file relative to
+              data_root.
+            - dataset_postfix (str, optional): Affects the postfix of the
+              resulting variable in the generated config. If specified, the
+              dataset variable will be named in the form of
+              ``{dataset_name}_{dataset_postfix}_{task}_{split}``. Defaults to
+              None.
+        val_anns (List[Dict], optional): A list of val annotation files
+            to appear in the base configs, similar to ``train_anns``. Defaults
+            to [].
+        test_anns (List[Dict], optional): A list of test annotation files
+            to appear in the base configs, similar to ``train_anns``. Defaults
+            to ``[dict(file='textdet_test.json')]``.
+        config_path (str): Path to the configs. Defaults to 'configs/'.
+    """
+
+    def __init__(
+        self,
+        data_root: str,
+        dataset_name: str,
+        train_anns: Optional[List[Dict]] = [
+            dict(ann_file='textdet_train.json')
+        ],
+        val_anns: Optional[List[Dict]] = [],
+        test_anns: Optional[List[Dict]] = [dict(ann_file='textdet_test.json')],
```

Comment:
```suggestion
        test_anns: Optional[List[Dict]] = [dict(ann_file='textdet_test.json', dataset_postfix='')],
```
same for test_ann and textrecog. It can provide an example intuitively.